geowebcache/core/src/main/java/org/geowebcache/storage/blobstore/memory/MemoryBlobStore.java

```diff
@@ -742,4 +757,51 @@ public boolean executeOperation(BlobStore store, Object... objs)
         public abstract boolean executeOperation(BlobStore store, Object... objs)
                 throws StorageException;
     }
+
+    @Override
+    public boolean deleteByParametersId(String layerName, String parametersId)
+            throws StorageException {
+        componentsStateLock.lock();
+        try {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Removing Layer: " + layerName);
+            }
+            // Remove the layer from the cacheProvider
+            cacheProvider.removeLayer(layerName);
```

Comment:
As I recall I tried to figure out a more targeted way to do this, but MemoryBlobStore didn't provide it and adding it would have required digging into the internals of both of the cache implementations.  This PR was far reaching enough as it is, and it seemed to me memory blob store being inherently more transient than other store types this was an acceptable compromise.